test/suites/integration/ami_test.go

```diff
@@ -84,6 +84,49 @@ var _ = Describe("AMI", func() {
 
 		env.ExpectInstance(pod.Spec.NodeName).To(HaveField("ImageId", HaveValue(Equal(customAMI))))
 	})
+	It("should support ami selector aws::name and aws::owners", func() {
+		output, err := env.EC2API.DescribeImages(&ec2.DescribeImagesInput{
+			ImageIds:[]*string{aws.String(customAMI)},
+		})
+		Expect(err).To(BeNil())
+		Expect(output.Images).To(HaveLen(1))
+
+		provider := awstest.AWSNodeTemplate(v1alpha1.AWSNodeTemplateSpec{AWS: v1alpha1.AWS{
+			SecurityGroupSelector: map[string]string{"karpenter.sh/discovery": settings.FromContext(env.Context).ClusterName},
+			SubnetSelector:        map[string]string{"karpenter.sh/discovery": settings.FromContext(env.Context).ClusterName},
+			AMIFamily:             &v1alpha1.AMIFamilyCustom,
+		},
+			AMISelector: map[string]string{"aws::name": *output.Images[0].Name, "aws::owners": *output.Images[0].ImageOwnerAlias},
```

Comment:
Can you add a test without owners and ensure it fails to provision?

Reply:
Owners gets defaulted to `amazon, self`, so it should theoretically work without the `owners`. I'll add a test with mismatching owner.

Reply:
Ah, alright, maybe a test with no owners and verify on the mock that owners is set?